src/sky.py

```diff
@@ -9,88 +9,64 @@
-            if (self.game_hour > 23):    # hours cycle every 24 in game hours
-                self.game_hour = 0
-
-            # Loop through each index of the current overlay color, then
-            # increment it by the above determined increments between noon and
-            # midnight
-            for index, value in enumerate(self.end_color):
-                if (self.game_hour >= 12):
-                    self.start_color[index] -= self.color_increment[index]
-                    if (self.start_color[index] < 0):
-                        self.start_color[index] = 0
-                else:
-                    self.start_color[index] += self.color_increment[index]
-                    if (self.start_color[index] > 255):
-                        self.start_color[index] = 255
-
-        self.full_surf.fill(self.start_color)
-        self.display_surface.blit(
-            self.full_surf, (0, 0), special_flags=pygame.BLEND_RGBA_MULT)
-
-    def set_time(self, hours, minutes):
-        self.game_hour = hours
-        self.game_minute = minutes
-        num_of_minutes = hours * 60 + minutes
+        self.game_time = game_time
 
-        # if number of minutes is less than noon, add increments to color
-        # if number of minutes is greater than noon, subtract increments to
         # color
-        if (num_of_minutes < 720):
-            new_color = [self.end_color[0], self.end_color[1],
-                         self.end_color[2]]         # darkest color is midnight
-            for index, value in enumerate(self.end_color):
-                new_color[index] += self.color_increment[index] * \
-                    num_of_minutes
+        self.colors = {
+            '6' : (160, 187, 255),
+            '12' : (255, 255, 255),
+            '18' : (255, 240, 234),
+            '20' : (255, 219, 203),
+            '24' : (38, 101, 189)
+        }
+        self.color = self.get_color()
+
+    def get_color(self):
+
+        # get time
+        hour, minute = self.game_time.get_time()
+        precise_hour = hour + minute / 60
+
+        # find nearest hours in self.colors
+        color_index = 0
+        for index in range(len(self.colors)):
+            color_hour = int(list(self.colors.keys())[index])
+            if precise_hour < color_hour:
+                color_index = index - 1
+                break
         else:
-            num_of_minutes -= 720
-            # brightest color is noon
-            new_color = [255, 255, 255]
-            for index, value in enumerate(self.end_color):
-                new_color[index] -= self.color_increment[index] * \
-                    num_of_minutes
+            color_index = -1
+
+        # start and end colors
+        start_color = list(self.colors.values())[color_index]
+        end_color = list(self.colors.values())[color_index + 1]
+        start_hour = int(list(self.colors.keys())[color_index])
+        end_hour = int(list(self.colors.keys())[color_index + 1])
```

Comment:
No need to regenerate both of these lists twice, nothing prevents you from storing them in a variable. In fact, that might be better, because of algorithmic complexity

Reply:
Solved